multinode_integration_tests/tests/verify_bill_payment_utils/utils.rs

```diff
@@ -0,0 +1,1003 @@
+    contract_owner_wallet: Wallet,
+    seed: Seed,
+}
+
+struct BlockchainInterfaces {
+    standard_blockchain_interface: Box<dyn BlockchainInterface>,
+    web3: Web3<Http>,
+}
+
+pub struct GlobalValues {
+    pub test_inputs: TestInputs,
+    pub blockchain_params: BlockchainParams,
+    pub now_in_common: SystemTime,
+    blockchain_interfaces: BlockchainInterfaces,
+}
+
+pub struct WholesomeConfig {
+    pub global_values: GlobalValues,
+    pub consuming_node: ConsumingNode,
+    pub serving_nodes: [ServingNode; 3],
+}
+
+pub struct DebtsSpecs {
+    debts: [Debt; 3],
+}
+
+impl DebtsSpecs {
+    pub fn new(node_1: Debt, node_2: Debt, node_3: Debt) -> Self {
+        let debts = [node_1, node_2, node_3];
+        Self { debts }
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct Debt {
+    pub balance_minor: u128,
+    pub age_s: u64,
+}
+
+impl Debt {
+    pub fn new(balance_minor: u128, age_s: u64) -> Self {
+        Self {
+            balance_minor,
+            age_s,
+        }
+    }
+
+    fn proper_timestamp(&self, now: SystemTime) -> SystemTime {
+        now.checked_sub(Duration::from_secs(self.age_s)).unwrap()
+    }
+}
+
+pub trait NodeProfile {
+    fn ui_port(&self) -> Option<u16>;
+
+    fn debt_specs(&self) -> Debt;
+
+    fn derivation_path(&self) -> String;
+
+    fn name(&self) -> String;
+
+    fn gas_price_opt(&self) -> Option<u64>;
+}
+
+impl NodeProfile for ConsumingNodeProfile {
+    fn ui_port(&self) -> Option<u16> {
+        self.ui_port_opt
+    }
+
+    fn debt_specs(&self) -> Debt {
+        panic!("This method should be called only by the serving Nodes.")
+    }
+
+    fn derivation_path(&self) -> String {
+        derivation_path(0, 1)
+    }
+
+    fn name(&self) -> String {
+        "ConsumingNode".to_string()
+    }
+
+    fn gas_price_opt(&self) -> Option<u64> {
+        self.gas_price_opt
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum ServingNodeByName {
+    ServingNode1 = 1,
+    ServingNode2 = 2,
+    ServingNode3 = 3,
+}
+
+impl NodeProfile for ServingNodeProfile {
+    fn ui_port(&self) -> Option<u16> {
+        self.ui_port_opt
+    }
+
+    fn debt_specs(&self) -> Debt {
+        self.debt
+    }
+
+    fn derivation_path(&self) -> String {
+        derivation_path(0, (self.serving_node_by_name as usize + 1) as u8)
+    }
+
+    fn name(&self) -> String {
+        format!("{:?}", self.serving_node_by_name)
+    }
+
+    fn gas_price_opt(&self) -> Option<u64> {
+        None
+    }
+}
+
+pub fn establish_test_frame(
+    test_inputs: TestInputs,
+) -> (MASQNodeCluster, GlobalValues, BlockchainServer) {
+    let now = SystemTime::now();
+    let cluster = match MASQNodeCluster::start() {
+        Ok(cluster) => cluster,
+        Err(e) => panic!("{}", e),
+    };
+    let blockchain_server = BlockchainServer::new("ganache-cli");
+    blockchain_server.start();
+    blockchain_server.wait_until_ready();
+    let server_url = blockchain_server.url().to_string();
```

Comment:
Hust by glance you figured out that these 4 lines (386...389) can be refactored out to a new function called `start_blockchain_server()`. Possibly there are more lines that you'd like to refactor out from this function. Give it a try. 😉